### PR TITLE
refactor: use `_reportLoss` function on losses to reduce impact

### PR DESCRIPTION
**Pull request by fubuloubu**

### CODE REVIEW

**Review by poolpitako — APPROVED** [review]

**Comment by fubuloubu** [comment]
cc @jmonteer can you take a look at this fix?

File: contracts/Vault.vy [hunk 1]
```diff
@@ -1029,12 +1046,15 @@ def withdraw(
...
 
             # Reduce the Strategy's debt by the amount withdrawn ("realized returns")
             # NOTE: This doesn't add to returns as it's not earned by "normal means"
-            self.strategies[strategy].totalDebt -= withdrawn + loss
-            self.totalDebt -= withdrawn + loss
+            self.strategies[strategy].totalDebt -= withdrawn
+            totalWithdrawn += withdrawn
+
+        # NOTE: Do this once at the end of the withdrawal queue to reduce `SSTORE` ops
+        self.totalDebt -= totalWithdrawn
```
Comment (jmonteer) [inline contracts/Vault.vy:1057]: May changing `totalDebt` here cause problems when recalculating the ratio at the moment of reducing our trust with the strategy in line 944 (`self._totalAssets()` uses `self._totalDebt`) if we withdraw from several strategies and some report losses?

this is, if we withdraw from several strategies and some (other than the first) report losses, they will be seeing a higher _totalAssets than they should, resulting in a wrong `ratio_change` calculation

am I missing something?

Comment (fubuloubu) [reply contracts/Vault.vy:1057]: Great point! I will remove the gas optimization.

**Review by dougstorm — APPROVED** [review]
LGTM!

**Pull request merged**